backend/src/main/java/ro/utcn/sd/agui/a1/controller/QuestionsController.java

```diff
@@ -0,0 +1,50 @@
+package ro.utcn.sd.agui.a1.controller;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+import ro.utcn.sd.agui.a1.dto.AnswerDTO;
+import ro.utcn.sd.agui.a1.dto.QuestionDTO;
+import ro.utcn.sd.agui.a1.entity.User;
+import ro.utcn.sd.agui.a1.service.QuestionManagementService;
+import ro.utcn.sd.agui.a1.service.QuestionVoteManagementService;
+import ro.utcn.sd.agui.a1.service.UserDetailsManagementService;
+
+import java.util.List;
+
+@RestController
+@RequiredArgsConstructor
+public class QuestionsController {
+    private final QuestionManagementService questionManagementService;
+    private final QuestionVoteManagementService questionVoteManagementService;
+    private final UserDetailsManagementService userDetailsManagementService;
+
+    @GetMapping("/questions")
+    public List<QuestionDTO> readAll() {
+        return questionManagementService.listAllQuestions();
+    }
+
+    @GetMapping("/questions/title/{filterTitle}")
+    public List<QuestionDTO> readQuestionByTitle(@PathVariable("filterTitle") String title) {
```

Comment:
`@RequestParam`s are more suitable for filtering criteria than `@PathParam`s: https://www.baeldung.com/spring-request-param